metacat-connector-pig/src/main/java/com/netflix/metacat/connector/pig/converters/PigTypeConverter.java

```diff
@@ -31,29 +31,43 @@
 import com.netflix.metacat.common.type.TypeUtils;
 import com.netflix.metacat.common.type.VarcharType;
 import lombok.NonNull;
+import org.apache.pig.backend.executionengine.ExecutionEngine;
+import org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.LocalExecType;
 import org.apache.pig.data.DataType;
+import org.apache.pig.impl.PigContext;
 import org.apache.pig.impl.logicalLayer.FrontendException;
 import org.apache.pig.impl.logicalLayer.schema.Schema;
-import org.apache.pig.impl.util.Utils;
 import org.apache.pig.newplan.logical.Util;
 import org.apache.pig.newplan.logical.relational.LogicalSchema;
+import org.apache.pig.parser.QueryParserDriver;
 
 import javax.annotation.Nonnull;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Properties;
 
 /**
  * Class to convert pig to canonical type and vice versa.
  */
 public class PigTypeConverter implements ConnectorTypeConverter {
     private static final String NAME_ARRAY_ELEMENT = "array_element";
+    private static final PigContext PIG_CONTEXT = new PigContext(new LocalExecType() {
+        private static final long serialVersionUID = -54152102366768171L;
+
+        @Override
+        public ExecutionEngine getExecutionEngine(final PigContext pigContext) {
+            return null;
+        }
+    }, new Properties());
 
     /**
      * {@inheritDoc}.
      */
     @Override
     public Type toMetacatType(@Nonnull @NonNull final String pigType) {
         try {
-            final LogicalSchema schema = Utils.parseSchema(pigType);
+            final LogicalSchema schema = new QueryParserDriver(PIG_CONTEXT,
+                "util", new HashMap<>()).parseSchema(pigType);
```

Comment:
Can we make this QueryParserDriver instance to be reused ( instead of making the PigContext reused )?

Reply:
I am not sure if it is threadsafe. There is no documentation. I have to look into the method to decide that.

Reply:
sound reasonable to me, if you have identified the pigContext object is the root not this QueryParserDriver object.